src/platform/Linux/bluez/BluezEndpoint.cpp

```diff
@@ -715,56 +715,36 @@ static void EndpointCleanup(BluezEndpoint * apEndpoint)
 {
     if (apEndpoint != nullptr)
     {
-        if (apEndpoint->mpOwningName != nullptr)
-        {
-            g_free(apEndpoint->mpOwningName);
-            apEndpoint->mpOwningName = nullptr;
-        }
-        if (apEndpoint->mpAdapterName != nullptr)
-        {
-            g_free(apEndpoint->mpAdapterName);
-            apEndpoint->mpAdapterName = nullptr;
-        }
-        if (apEndpoint->mpAdapterAddr != nullptr)
-        {
-            g_free(apEndpoint->mpAdapterAddr);
-            apEndpoint->mpAdapterAddr = nullptr;
-        }
-        if (apEndpoint->mpRootPath != nullptr)
-        {
-            g_free(apEndpoint->mpRootPath);
-            apEndpoint->mpRootPath = nullptr;
-        }
-        if (apEndpoint->mpAdvPath != nullptr)
-        {
-            g_free(apEndpoint->mpAdvPath);
-            apEndpoint->mpAdvPath = nullptr;
-        }
-        if (apEndpoint->mpServicePath != nullptr)
-        {
-            g_free(apEndpoint->mpServicePath);
-            apEndpoint->mpServicePath = nullptr;
-        }
+        g_free(apEndpoint->mpOwningName);
+        g_free(apEndpoint->mpAdapterName);
+        g_free(apEndpoint->mpAdapterAddr);
+        g_free(apEndpoint->mpRootPath);
+        g_free(apEndpoint->mpAdvPath);
+        g_free(apEndpoint->mpServicePath);
+        if (apEndpoint->mpObjMgr != nullptr)
+            g_object_unref(apEndpoint->mpObjMgr);
```

Comment:
There is some issue with receiving some signal after the object has been destroyed (the crash is in the glib itself).

Adding blocking comment until further investigated.

Reply:
The issue was with `EndpointCleanup` being called on a non-glib thread, but signals were dispatched on glib thread - race condition leading to crash. Fixed by latest commit.